app/components/post/index.js

```diff
@@ -54,6 +57,21 @@ function mapStateToProps(state, ownProps) {
         }
     }
 
+    const previousPost = ownProps.previousPostId && getPost(state, ownProps.previousPostId);
+    if (previousPost) {
```

Comment:
can you move all this block to a selector? that way this will be memoized and won't be executed on every state change for every post

Reply:
actually this is pulled from a selector from mattermost-redux, because mobile version is not using the selector webapp uses(makeGetPosts).
if you want a cleanup, migrating this to makeGetPosts* should be a way to go
(i am author of this piece here)

Reply:
I kind of repeated the pattern of pulling code from that selector the same way isFirstReply=xxx here is copied

Reply:
What @koxen said is pretty much why I had it inside the mapStateToProps in the first place, because the isFirstReply was in there as well. I still moved it out of there now, though.
Also when merging the commit the credit I had placed there for @koxen's snippet got lost. Let me know if you want me to add you as a co-author to the commit ;)